thunder/tests/test_dynamo.py

```diff
@@ -29,12 +30,125 @@ def func(x):
 
     # out should have grad_fn and its name should be ThunderFunctionBackward
     assert out.grad_fn is not None
-    assert out.grad_fn.name() == "ThunderFunctionBackward"
+    if not dynamic:
+        # If dynamic, while trying to execute `x + 1`, we fail with
+        # "s0 had an unexpected type <class 'torch.SymInt'>. Supported types are (<class 'int'>, <class 'thunder.core.baseutils.NumberProxyInterface'>)")
```

Comment:
Where does this error originate from?

Reply:
Have fixed this as per discussion, now we get the concrete value from the tensor and create a FakeTensor with concrete values to create the new Tensor proxies.